components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,556 @@
+	minPowerPct float64
+	// the maximum power that can be set for the motor
+	maxPowerPct float64
+	// the freewheel RPM of the motor
+	maxRPM float64
+
+	// A manager to ensure only a single operation is happening at any given time since commands could overlap on the serial port
+	opMgr operation.SingleOperationManager
+}
+
+// Config adds DimensionEngineering-specific config options.
+type Config struct {
+	// path to /dev/ttyXXXX file
+	SerialPath string `json:"serial_path"`
+
+	// The baud rate of the controller
+	BaudRate int `json:"serial_baud_rate,omitempty"`
+
+	// Valid values are 128-135
+	SerialAddress int `json:"serial_address"`
+
+	// Valid values are 1/2
+	MotorChannel int `json:"motor_channel"`
+
+	// Flip the direction of the signal sent to the controller.
+	// Due to wiring/motor orientation, "forward" on the controller may not represent "forward" on the robot
+	DirectionFlip bool `json:"dir_flip,omitempty"`
+
+	// A value to control how quickly the controller ramps to a particular setpoint
+	RampValue int `json:"controller_ramp_value,omitempty"`
+
+	// The maximum freewheel rotational velocity of the motor after the final drive (maximum effective wheel speed)
+	MaxRPM float64 `json:"max_rpm,omitempty"`
+
+	// The name of the encoder used for this motor
+	Encoder string `json:"encoder,omitempty"`
+
+	// The lowest power percentage to allow for this motor. This is used to prevent motor stalls and overheating. Default is 0.0
+	MinPowerPct float64 `json:"min_power_pct,omitempty"`
+
+	// The max power percentage to allow for this motor. Default is 0.0
+	MaxPowerPct float64 `json:"max_power_pct,omitempty"`
+
+	// The number of ticks per rotation of this motor from the encoder
+	TicksPerRotation int `json:"ticks_per_rotation,omitempty"`
+
+	// TestChan is a fake "serial" path for test use only
+	TestChan chan []byte `json:"-,omitempty"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (cfg *Config) Validate(path string) error {
+	if cfg.SerialPath == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "serial_path")
+	}
+	if cfg.MotorChannel < 1 || cfg.MotorChannel > 2 {
+		return utils.NewConfigValidationError(path, errors.New("invalid channel, acceptable values are 1 and 2"))
+	}
+	if cfg.SerialAddress < 128 || cfg.SerialAddress > 135 {
+		return utils.NewConfigValidationError(path, errors.New("invalid address, acceptable values are 128 thru 135"))
+	}
+	if cfg.BaudRate != 2400 && cfg.BaudRate != 9600 && cfg.BaudRate != 19200 && cfg.BaudRate != 38400 && cfg.BaudRate != 115200 {
+		return utils.NewConfigValidationError(path, errors.New("invalid baud_rate, acceptable values are 2400, 9600, 19200, 38400, 115200"))
+	}
+	if cfg.MinPowerPct < 0.0 || cfg.MinPowerPct > cfg.MaxPowerPct {
+		return utils.NewConfigValidationError(path, errors.New("invalid min_power_pct, acceptable values are 0 to max_power_pct"))
+	}
+	if cfg.MaxPowerPct > 100.0 {
```

Comment:
Other motors in the repo only support powers in [-1;1] is there a reason to enforce a [0;100] range?